### PR TITLE
Toolkit feature backend

### PR DESCRIPTION
**Pull request by luciferlinx101**
Toolkit feature backend changes consisting :
- Tool's new architecture
- introduction of toolkit  
- tool configuration at the toolkit level
- Model Unit tests
- Helper method unit tests
- Changes in current tools as per new architecture and toolkit.
- Introduced the concept of tool manager and managing tools using tools.json for external tools
- Introducing marketplace for tools

### CODE REVIEW

File: tool_manager.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import os
+import requests
+import zipfile
+import json
+
+# def parse_github_url(github_url):
```
Comment (TransformerOptimus) [inline tool_manager.py:6]: remove these comments if unused.

File: superagi/models/agent_config.py [hunk 2]
```diff
@@ -30,3 +32,16 @@ def __repr__(self):
 
         """
         return f"AgentConfiguration(id={self.id}, key={self.key}, value={self.value})"
+
```
Comment (TransformerOptimus) [inline superagi/models/agent_config.py:35]: add class method or static method

File: superagi/models/tool.py [hunk 3]
```diff
@@ -32,7 +35,39 @@ def __repr__(self):
...
+                        file_name=file_name,
+                        toolkit_id=toolkit_id)
+            session.add(tool)
+
+        session.commit()
+        session.flush()
+        return tool
+
+    @staticmethod
+    def delete_tool(session, tool_name):
```
Comment (TransformerOptimus) [inline superagi/models/tool.py:65]: make it class method

File: superagi/models/toolkit.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        return cls(
+            id=data['id'],
+            name=data['name'],
+            description=data['description'],
+            show_toolkit=data['show_toolkit'],
+            organisation_id=data['organisation_id']
+        )
+
+    @staticmethod
+    def add_or_update(session, name, description, show_toolkit, organisation_id, tool_code_link):
```
Comment (TransformerOptimus) [inline superagi/models/toolkit.py:59]: make it class method

File: superagi/models/toolkit.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        response = requests.get(
+            marketplace_url + f"/toolkits/marketplace/{search_str}/{toolkit_name}",
+            headers=headers, timeout=10)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            return None
+
+    @staticmethod
+    def get_toolkit_from_name(session, toolkit_name):
```
Comment (TransformerOptimus) [inline superagi/models/toolkit.py:111]: class method

**Review by TransformerOptimus — APPROVED** [review]

File: tool_manager.py (hunk 1 above)
Comment (Tdmain) [reply tool_manager.py]: You good sorry if you feel like I took yo long I got to much on my plate you did good, real good.

**Pull request merged**